Updates IR Swift configuration docs to add a note (#83341)
openshift/openshift-docs: modules/registry-configuring-registry-storage-swift-trust.adoc

[id="registry-configuring-registry-storage-swift-trust_{context}"]

> = Configuring the Image Registry Operator to trust Swift storage

Reviewer: The title of this part need update too.

Author: @xiuwang What do you think? "Configuring the Image Registry for Red Hat OpenStack Platform Swift Storage"

Thanks!

Reviewer: How about "Manage the Image Registry Operator Redirects" ?

Author: @xiuwang Slightly changed, but updated. PTAL! Thanks!

> You can configure the Image Registry Operator to trust {rh-openstack-first} Swift storage. This is an optional procedure depending on how your clients trust storage certificates.

Reviewer: This description still doesn't seem quite right to me.
It is not to let  Image Registry Operator to trust storage, according to https://bugzilla.redhat.com/show_bug.cgi?id=1810461#c11, it should be the clients
In addition, how we define the clients here?

Reviewer 2: 👆🏼  +1

Reviewer 2: Clients here could be for example a developer laptop's, but also a build in the cluster (I think builds is what this was first documented for).

> By disabling redirect, you can configure the Image Registry Operator to control whether clients such as {product-title} cluster builds or external systems like developer machines are redirected to pull images directly from {rh-openstack-first} Swift storage. This configuration is optional and depends on whether the clients trust the storage's SSL/TLS certificates.

Reviewer 3: ```suggestion
By disabling redirects, you can configure the Image Registry Operator to control whether clients such as {product-title} cluster builds or external systems like developer machines are redirected to pull images directly from {rh-openstack-first} Swift storage. This configuration is optional and depends on whether the clients trust the storage SSL or TLS certificates.
```

[NOTE]
====

> In situations where clients to not trust the storage certificate, the `disableRedirect` option can be set to `true` to proxy traffic through the image registry. Consequently, however, the image registry requires significantly more resources, especially network bandwidth, to handle the increased load. 

Reviewer 2: Nit:
```suggestion
In situations where clients to not trust the storage certificate, the `disableRedirect` option can be set to `true` to proxy traffic through the image registry. Consequently, however, the image registry may require significantly more resources, especially network bandwidth, to handle the increased load. 
```

If the image registry is not heavily used and the images in storage are not huge then disabling redirect might not cause any performance impact. So "may" seems more accurate.

> In situations where clients to not trust the storage certificate, the `disableRedirect` option can be set to `true` to proxy traffic through the image registry. Consequently, however, the image registry might require more resources, especially network bandwidth, to handle the increased load. 

Reviewer 3: ```suggestion
In situations where clients to not trust the storage certificate, setting the `disableRedirect` option to `true` proxies traffic through the image registry. Consequently, the image registry might require more resources, especially network bandwidth, to handle the increased load. 
```

Alternatively, if clients trust the storage certificate, the registry can allow redirects. This reduces resource demand on the registry itself. 

Some users might prefer to configure their clients to trust their self-signed certificate authorities (CAs) instead of disabling redirects. If you are using a self-signed CA, you must decide between trusting the custom CAs or disabling redirects.
====
// to allow the client to pull the image layers from the image registry rather than from links directly from Swift.

> 

Reviewer 3: Does this require `oc` access as admin as a prereq?

.Procedure

> * From a command line, enter the following command to change the value of the `spec.disableRedirect` field in the `config.imageregistry` object to `true`. This ensures that the image registry will proxy traffic instead of relying on Swift storage.

Reviewer 3: ```suggestion
* To ensures that the image registry proxies traffic instead of relying on Swift storage, change the value of the `spec.disableRedirect` field in the `config.imageregistry` object to `true` by running the following command:
```
